File: duolingo_rum/Models/Language.cs
using System;
using System.Collections.Generic;

namespace duolingo_rum.Models;

public partial class Language
{
    public int Id { get; set; }

    public string Code { get; set; } = null!;

    public string Name { get; set; } = null!;

    public string NativeName { get; set; } = null!;

    public string? FlagEmoji { get; set; }

    public bool? IsActive { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<LearningSession> LearningSessions { get; set; } = new List<LearningSession>();

    public virtual ICollection<Topic> Topics { get; set; } = new List<Topic>();

    public virtual ICollection<User> UserNativeLanguages { get; set; } = new List<User>();

    public virtual ICollection<User> UserTargetLanguages { get; set; } = new List<User>();

    public virtual ICollection<Word> Words { get; set; } = new List<Word>();
}

File: android-app/app/src/main/java/com/linguaai/app/models/Language.kt
package com.linguaai.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Language(
    val id: Int = 0,
    val code: String = "",
    val name: String = "",
    @SerialName("native_name")
    val nativeName: String = "",
    @SerialName("flag_emoji")
    val flagEmoji: String? = null,
    @SerialName("is_active")
    val isActive: Boolean? = true,
    @SerialName("created_at")
    val createdAt: String? = null
)
